speech/samples/v1p1beta1/samplegen_read_and_write_files.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# DO NOT EDIT! This is a generated sample ("Request",  "samplegen_read_and_write_files")
+
+# To install the latest published package dependency, execute the following:
+#   pip install google-cloud-speech
+
+# sample-metadata
+#   title: Read binary file into bytes field & write string in response to file
+#   description: Read binary file into bytes field & write string in response to file
+#   usage: python3 samples/v1p1beta1/samplegen_read_and_write_files.py
+
+# [START samplegen_read_and_write_files]
+from google.cloud import speech_v1p1beta1
+from google.cloud.speech_v1p1beta1 import enums
+import io
+
+
+def sample_recognize():
+    """Read binary file into bytes field & write string in response to file"""
+
+    client = speech_v1p1beta1.SpeechClient()
+
+    encoding = enums.RecognitionConfig.AudioEncoding.MP3
+    config = {"encoding": encoding}
+
+    # The bytes from this file will be read into `content`
+    with io.open("path/to/file.mp3", "rb") as f:
```

Comment:
Yes, it should be.

Reply:
Actually, we have a convention of using `"resources/<something>"` so this will work for users locally.

We've received a surprising number of bug reports from users running our samples and reporting errors like "I got a File Not Found cat.png!" so we [ship a folder of resources](https://github.com/googleapis/googleapis/blob/master/google/cloud/speech/sample_resources.yaml) for [each library](https://github.com/googleapis/google-cloud-python/tree/master/speech/samples/resources) (_from the standard `gs://cloud-samples-data` bucket_). Usually this should be .csv's and the like, but there are 3 products which work on binary files ~> Vision (images), Speech (audio), Video (videos). Sample authors for those products should be very aware that any resources they add will end up in client library repos, so use sparingly.

Most products which accept `bytes` also accept GCS file paths which is generally favorable. So most samples tend to use GCS paths, but at least 1 sample per product should generally show developers how to Hello World call the API using an existing local file.

Reply:
^---- follows the conventions of `python-docs-samples` (search for `resources/` directories)

Reply:
If we follow the convention that most users are copy/pasting-ing samples from the docs site though, they won't have access to this file. It feels like it would make the most sense to have these be an argument with a reasonable default that the user can easily override.